#50 Terminology (#55)
Inbibo/shift_docs: docs/getting_started/basics/terminology.md

# Terminology

This page assembles a list of definitions for some of Shift's main concepts.

## Workflow

> A workflow is a directed acyclic graph (DAG) that builds the relationship between the execution of its nodes. A workflow is composed of interconnected nodes and defines an execution process.

Reviewer: ```suggestion
A workflow is a directed acyclic graph (DAG) that builds the causal relationship between the execution of its nodes. A workflow is composed of interconnected nodes and defines the global execution logic of your automation.
```

> ## Node
> Nodes are the graphic block units that build up the workflow. Each node is associated with an operator, which defines the execution logic of the node and is connected to other nodes through plugs.

Reviewer: ```suggestion
## Node

Nodes are the graphic block units that build up the workflow. Each node is associated with an operator, which defines the execution logic of the node and is connected to other nodes through plugs.
```

## Plug

> A plug represents an input or output of a node. Plugs have a type associated based their compatibility with different Python data types. Plug values can be set manually via the inspector widget or gotten from a connected plug.

Reviewer: ```suggestion
A plug represents an input or output of a node. Plugs have associated types based on the Python object that they should contain. Plug values can be set manually via the inspector widget or propagated from a connected plug.
```

Author: Done

## Connection

> A connection is the graphical representation of two connected plugs and shows the direction of the data propagation. Connections between plugs from different nodes are what define the execution stack of the workflow. 

Reviewer: ```suggestion
A connection is the graphical representation of the relationship between two plugs and shows the direction of the data propagation. Connections between plugs from different nodes are what defines the execution stack of the workflow. 
```

## Execution Stack

> The execution stack refers to the order in which nodes will be executed. The execution stack is closely related to the workflow configuration, although it can be more or less extended and include repeated nodes under the presence of iterators or conditions. 

Reviewer: ```suggestion
The execution stack refers to the order in which nodes will be executed. The execution stack is closely related to the workflow configuration. It can be a linear execution list of nodes in the workflow or dynamically expand or contract under the presence of iterators or conditionals. 
```

## Board

> Refers to the workspace in the Shift UI where the workflow is constructed.

Reviewer: ```suggestion
The term Board refers to the workspace in the Shift UI where the graphical counterpart of the workflow is constructed. It is the master object holding all information on the graphic items in the interface.
```

Author: Done

## Inspector

> The inspector is the UI widget through which the name, inputs, and output values of a node can be inspected. It is also where values can be set manually in input nodes. 

Reviewer: ```suggestion
The inspector is the UI widget through which the name, inputs, and output values of a node can be inspected or manually modified.
```

## Operator

> An operator is the scripted execution block that performs a specific process given a set of input data if required. From a programming perspective, an operator can be seen as a function that takes some input data and evaluates it to produce some output data.

Reviewer: ```suggestion
An operator is the logical counterpart of a node and performs a specific process given a set of input data, if required. From a programming perspective, an operator can be seen as a function that takes some input data and evaluates it to produce some output data.
```

## Catalog

> A catalog is a collection of operators that usually share a common context.  Users can create their catalogs to organize their custom operators.

Reviewer: ```suggestion
A catalog is a collection of operators that usually share a common context.  Users can create their own catalogs to integrate their custom operators into Shift.
```

## Plugin

> A plugin is a custom-based UI that can interact with Shift to display additional information, perform actions, or extend the behavior of the application.

Reviewer: ```suggestion
A plugin is a custom-based UI that can interact with Shift to display additional information, perform actions, or extend the behavior of the application.

```

# Terminology

This page assembles a list of definitions for some of Shift's main concepts.

## Workflow

A workflow is a directed acyclic graph (DAG) that builds the casual relationship between the execution of its nodes. A workflow is composed of interconnected nodes and defines the global execution logic of the automation process.

## Node

Nodes are the graphic block units that build up the workflow. Each node is associated with an operator, which defines the execution logic of the node and is connected to other nodes through plugs.

## Plug

A plug represents an input or output of a node. Plugs have associated types based on the Python object that they should contain. Plug values can be set manually via the inspector widget or propagated from a connected plug.

## Connection

> A connection is the graphical representation of the relationship between two plugs and shows the direction of the data propagation. Connections between plugs from different nodes are what defines the execution stack of the workflow. 

Reviewer 2: ```suggestion
A connection is the graphical representation of the relationship between two plugs and shows the direction of the data propagation. Connections between plugs from different nodes define the execution stack of the workflow. 
```

## Execution Stack

The execution stack refers to the order in which nodes will be executed. The execution stack is closely related to the workflow configuration. It can be a linear execution list of nodes in the workflow or dynamically expand or contract under the presence of iterators or conditionals. 

## Board

> The board refers to the workspace in the Shift UI where the graphical counterpart of the workflow is build. It is the master object holding all information on the graphic items in the interface.

Reviewer 2: I would change the `master` term to use `main`, as happened to github with the branch naming:
```suggestion
The board refers to the workspace in the Shift UI where the graphical counterpart of the workflow is build. It is the main object holding all information on the graphic items in the interface.
```

## Inspector

> The inspector is the UI widget through which the name, inputs, and output values of a node can be inspected or manually modified.

Reviewer 2: ```suggestion
The inspector is the UI widget through which the name of the node, inputs, and output plug values of a node can be inspected or manually modified.
```
